packages/components/atoms/f-filter-pill/src/components/FilterPill.vue

```diff
@@ -1,29 +1,184 @@
+            }]"
+        data-test-id="filter-item">
+        <input
+            :id="`filterPillToggle-${inputId}`"
+            type="checkbox"
+            class="is-visuallyHidden"
+            :class="$style['c-filterPill-checkbox']"
+            data-test-id="filter-pill-input"
+            :tabindex="0"
+            :disabled="isToggleDisabled"
+            @change="toggleFilter">
+        <label
+            :class="$style['c-filterPill-label']"
+            :for="`filterPillToggle-${inputId}`"
+            data-test-id="filter-pill-label"
+            :tabindex="-1">
+            <tick-icon :class="$style['c-filterPill-icon']" />
+            <span
+                :class="$style['c-filterPill-text']"
+                data-test-id="filter-pill-text">
+                {{ displayText }}
+            </span>
+            <span
+                :class="$style['c-filterPill-number']"
+                data-test-id="filter-pill-number">
+                {{ displayNumber }}
+            </span>
+        </label>
     </div>
 </template>
 
 <script>
+import { TickIcon } from '@justeat/f-vue-icons';
+
 export default {
     name: 'FilterPill',
-    components: {},
+    components: {
+        TickIcon
+    },
     props: {
+        inputId: {
+            type: String,
+            default: null
+        },
+        toggleValue: {
+            type: String,
+            default: null
+        },
+        isSelected: {
+            type: Boolean,
+            default: false
+        },
+        isDisabled: {
+            type: Boolean,
+            default: false
+        },
+        displayText: {
+            type: String,
+            default: null
+        },
+        displayNumber: {
+            type: Number,
+            default: null
+        }
+    },
+    data () {
+        return {
+            isToggleSelected: false,
+            isToggleDisabled: false
+        };
+    },
+    computed: {},
+    watch: {
+        isSelected (newValue) {
+            if (newValue !== this.isToggleSelected) {
+                this.isToggleSelected = newValue;
+            }
+        },
+        isDisabled (newValue) {
+            if (newValue !== this.isToggleDisabled) {
+                this.isToggleDisabled = newValue;
+            }
+        }
+    },
+    created () {
+        this.isToggleSelected = this.isSelected;
+        this.isToggleDisabled = this.isDisabled;
+    },
+    methods: {
+        toggleFilter () {
+            this.isToggleSelected = !this.isToggleSelected;
+
+            this.$nextTick(() => {
+                this.$emit('toggle', this.toggleValue);
+            });
+        }
     }
 };
 </script>
 
 <style lang="scss" module>
+$filter-pill-selected: $color-content-positive;
+$filter-pill-checkbox-width: 16px;
+
+@mixin ellipsis() {
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+}
+
 .c-filterPill {
+    outline: 1px solid $color-grey-30;
```

Comment:
@ashleynolan this might be a bit of a wild idea but would it be possible for us to set up a linter / PR check soft warning when using pie tokens that start with `$color` that could encourage to look for an alias?

Reply:
@jamieomaguire I imagine this is v possible in stylelint. Would be cool to add a ticket to do this in the future for sure 👍

Reply:
Sounds good! Yeah stylelint sounds promising for this :)